### PR TITLE
Download file as stream and add cancel button

### PR DESCRIPTION
**Pull request by cameronmarsh**
smartcitiesdata/smartcitiesdata#790

### CODE REVIEW

File: apps/andi/assets/js/app.js [hunk 1]
```diff
@@ -35,15 +35,17 @@ Hooks.showSnackbar = {
...
+            var file = this.el.files[0];
 
             fileToText(this.el.files[0]).then(fileAsText => {
                 this.pushEvent("file_upload", {
                     file: fileAsText,
                     fileType: file["type"],
                     fileSize: file["size"]
                 });
+            }, reason => {
+                reason == "cancel" & this.pushEvent("file_upload_cancelled");
```
Comment (bbrewer-forge) [inline apps/andi/assets/js/app.js:48]: wait, what does `&` do here? Is it an XOR?

Comment (cameronmarsh) [reply apps/andi/assets/js/app.js:48]: no its basically a shorthand if statement

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 2]
```diff
@@ -80,12 +80,15 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
-                    <%= ErrorHelpers.error_tag(f, :schema_sample, bind_to_input: false) %>
+                  <div class="file-input-button--<%= loader_visibility %>">
+                    <div class="file-input-button">
+                      <%= label(f, :schema_sample, "Upload data sample", class: "label") %>
+                      <%= file_input(f, :schema_sample, phx_hook: "readFile", accept: "text/csv, application/json") %>
+                      <%= ErrorHelpers.error_tag(f, :schema_sample, bind_to_input: false) %>
+                    </div>
                   </div>
 
+                  <button type="button" id="reader-cancel" class="file-upload-cancel-button file-upload-cancel-button--<%= loader_visibility %> btn">Cancel</button>
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:91]: How does this flow down to the "change" event? Is it part of the `file_input` internals?

Comment (cameronmarsh) [reply apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:91]: There is an event listener in the `app.js` file, where the FileReader is being defined

File: apps/andi/assets/js/app.js (hunk 1 above)
Comment (cameronmarsh) [reply apps/andi/assets/js/app.js]: yeah that should be `&&`

**Pull request merged**